src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -371,5 +372,56 @@ public CloudSdk build() {
           startListeners, runDevAppServerWaitSeconds);
     }
 
+    /**
+     * Attempt to find the Google Cloud SDK in various places.
+     * 
+     * @return the path to the root of the Google Cloud SDK
+     * @throws AppEngineException if not found
+     */
+    @Nonnull
+    private Path discoverSdkPath() {
+      for (CloudSdkResolver resolver : getResolvers()) {
+        try {
+          Path discoveredSdkPath = resolver.getCloudSdkPath();
+          if (discoveredSdkPath != null) {
+            return discoveredSdkPath;
+          }
+        } catch (Exception ex) {
```

Comment:
No, catching Exception is highly discouraged. (See Bloch, Effective Java). Catching Throwable is worse. In some cases it's OK to catch RuntimeException, but I'm not sure this is one of them. Usually you only want to do that when you don't know what code might be invoked. Here we should only catch what getCloudSdkPath might plausibly throw.